### ISSUE TITLE
Update S2436: Ignore for inherited parameters

### ISSUE THREAD

**Issue by NoxArt** [SonarSource/sonar-dotnet#2114]
### Description
Situation when implementing an abstract class from an external framework that contains >= DefaultMaxNumberOfGenericParametersInClass or -InMethod and the implementing class supplies more specific types than allowed to fullfil what the parent requests.

It should be possible to ignore the specific section in a file, this is to present suggestion / open discussion that this case would be ignored in general, or at least as an option (not sure if bool parameters are supported) ... or maybe tell us that what we're doing is a bad practice anyway.

But we still agree with the rule in general, newly created classes that we (authors of the project running SonarQube) are able to control shouldn't go overboard with the number of generic parameters.

### Repro steps

    // with DefaultMaxNumberOfGenericParametersInClass = 2 for example
    namespace SomeFramework {
        abstract class Parent<T1,T2,T3>{ }
    }
    namespace OurProject {
        class Child: Parent<ISomeInterface1,ISomeInterface2,SomeImplementation1> { }
    }

### Expected behavior

Issue S2436 will not be reported for the Child class (exceeded DefaultMaxNumberOfGenericParametersInClass when set to 2)

### Actual behavior

Issue S2436 is reported for the Child class

### Known workarounds

Suppress error in the code manually each time (or close the issue as false positive in the website)

### Related information

* SonarC# Version 6.7.3 (I've checked the source code of the rule and it seems to be still the same)

**Comment by Evangelink** [comment]
Hi @NoxArt,

Thanks for the feedback! I have tried the following snippet (based on your code sample) on the latest version of SonarC# and I have the correct behavior. Could you let me know if you meant something else?

```csharp
namespace MyLib
{
    public abstract class FrameworkBaseClass<T1, T2, T3> // Noncompliant
    {

    }
}

namespace TheProject
{
    using MyLib;

    public class Impl : FrameworkBaseClass<int, double, bool>
    {

    }
}
```

**Comment by NoxArt** [comment]
Thank you for reply. Yes, this is what I meant - but it seems I made a mistake, looks like our classes also introduced some new ones which triggered the rule, sorry for taking your time
